lib/cfonb/operation.rb

```diff
@@ -20,6 +24,34 @@ class Operation
 
     attr_accessor(*BASE_ATTRIBUTES)
 
+    def self.parse(input)
+      current_operation = nil
+
+      input.each_line do |line|
+        (line.size / 120).times do |index|
+          start = index * 120
+          finish = start + 119
+
+          line = CFONB120::LineParser.parse(line[start..finish])
+
+          case line.code
+          when '04'
+            raise AlreadyDefinedOperationError if current_operation
+
+            current_operation = new(line)
+          when '05'
+            raise UnstartedOperationError unless current_operation
+
+            current_operation.merge_detail(line)
+          else
+            raise UnhandledLineCodeError
+          end
+        end
+      end
+
+      current_operation
+    end
```

Comment:
Can we instead define a `def self.parse_operation` or equivalent in `lib/cfonb.rb` to keep the public interface there.
Then move the parsing logic to a `lib/cfonb/operation_parser.rb` or within `lib/cfonb/parser.rb`?